README.md

```diff
@@ -1,17 +1,242 @@
+   ``` cs
+   using ReviewService;
+   using ReviewService.Abstractions;
+
+   var reviewService = new ReviewService<ReviewSettings>(
+       logger: null,
+       reviewPrompter: new ReviewPrompter(logger: null),
+       reviewSettingsSource: new ReviewSettingsSource(),
+       reviewConditionsBuilder: ReviewConditionsBuilder.Empty().MinimumPrimaryActionsCompleted(1)
+   )
+   ```
+
+7. Track and update review settings that are used by review conditions to know if a review should be prompted (let's assume the service was injected and we are using it inside a class).
+   ``` cs
+   using ReviewService.Abstractions;
+
+   private readonly IReviewService<ReviewSettings> _reviewService;
+
+   public async Task DoPrimaryAction(CancellationToken ct)
+   {
+       // Do Primary Action.
+
+       // Track this action.
+       _reviewService.TrackPrimaryActionCompleted(ct)
+   }
+   ```
+
+6. Use the service to request review (let's assume the service was injected and we are using it inside a class).
+    ``` cs
+   using ReviewService.Abstractions;
+
+   private readonly IReviewService<ReviewSettings> _reviewService;
+
+   public async Task DoMeaningfulTask(CancellationToken ct)
+   {
+       // Do Meaningful Task.
+
+       // Check if all conditions are satisfied and prompt for review if they are.
+       await _reviewService.TryRequestReview(ct);
+   }
+   ```
+
+   <img src="docs/review_prompt_android.gif" alt="Review Prompt Android" width="250">
+   <img src="docs/review_prompt_ios.gif" alt="Review Prompt iOS" width="250">
+
+### Using Dependency Injection
+
+Here is a simple code that does dependency injection using `Microsoft.Extensions.DependencyInjection` and `Microsoft.Extensions.Hosting`.
+
+``` cs
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using ReviewService;
+using ReviewService.Abstractions;
+
+var host = new HostBuilder()
+    .ConfigureServices(serviceCollection => serviceCollection
+        .AddSingleton<IReviewSettingsSource<ReviewSettingsCustom>, ReviewSettingsSource>()
+        .AddTransient(s => ReviewConditionsBuilder.Default<ReviewSettingsCustom>())
+        .AddSingleton<IReviewService<ReviewSettingsCustom>, ReviewService<ReviewSettingsCustom>>()
+    )
+    .Build();
+```
 
 ## Features
 
-{More details/listing of features of the project}
+The previous setup was pretty basic. Let's see what else we can do!
+
+### Default Review Conditions
+
+If you want to use our default review conditions, you can use `ReviewConditionsBuilder.Default()` and and pass it to the `ReviewService` constructor, or inject it as a transient when using dependency injection. It's also possible to pass `null` to the constructor, it will use `ReviewConditionsBuilder.Default()` by itself. Please note that our review conditions are also generic, so they can be used with custom review settings too.
```

Comment:
You can remove the following:

It's also possible to pass `null` to the constructor, it will use `ReviewConditionsBuilder.Default()` by itself.